cloud-gateway-service/src/test/java/org/zowe/apiml/cloudgatewayservice/service/CentralApimlInfoMapperTest.java

```diff
@@ -0,0 +1,143 @@
+        void shouldReturnEmptyServicesWhenServiceListContainsNulls() {
+            ApimlInfo info = centralApimlInfoMapper.buildApimlServiceInfo("apiml1", asList(null, null, null));
+
+            assertThat(info.getApimlId()).isEqualTo("apiml1");
+            assertThat(info.getServices()).isEmpty();
+        }
+    }
+
+    @Nested
+    class WhenMappingServiceInfo {
+
+        private ServiceInfo serviceOne;
+        private ServiceInfo.Apiml apiml;
+        private ServiceInfo.ApiInfoExtended apiInfo;
+
+        @BeforeEach
+        public void setUp() {
+            centralApimlInfoMapper.metadataKeysAllowList = Sets.set("zos.sysname", "zos.system", "zos.sysplex", "zos.cpcName", "zos.zosName", "zos.lpar");
+
+            apiml = new ServiceInfo.Apiml();
+            apiInfo = ServiceInfo.ApiInfoExtended.builder().apiId("zowe.apiml.apicatalog").build();
+            apiml.setApiInfo(singletonList(apiInfo));
+            ServiceInfo.Instances instance = ServiceInfo.Instances.builder()
+                    .customMetadata(Maps.of("zos.sysname", "MD20",
+                            "zos.sysplex", "PLEXM2",
+                            "zos.jobid", "TST71112"
+                    )).build();
+
+            //Well decorated service
+            Map<String, ServiceInfo.Instances> instances = Maps.of("dummy:host:14", instance);
+            serviceOne = ServiceInfo.builder().serviceId("serviceId")
+                    .status(InstanceInfo.InstanceStatus.UP)
+                    .apiml(apiml)
+                    .instances(instances)
+                    .build();
+        }
+
+        @Test
+        void shouldMapServiceInfoWithFilteredMetadata() {
+            ApimlInfo info = centralApimlInfoMapper.buildApimlServiceInfo("apiml1", singletonList(serviceOne));
+
+            assertThat(info.getApimlId()).isEqualTo("apiml1");
+            CentralServiceInfo centralService = info.getServices().get(0);
+
+            assertThat(centralService.getServiceId()).isEqualTo("serviceId");
+            assertThat(centralService.getStatus()).isEqualTo(InstanceInfo.InstanceStatus.UP);
+            assertThat(centralService.getApiId()).containsOnly("zowe.apiml.apicatalog");
+            assertThat(centralService.getCustomMetadata()).containsOnlyKeys("zos.sysname", "zos.sysplex");
+        }
+
+        @Test
+        void shouldMapServiceInfoWithMultipleAPIs() {
+            ServiceInfo.ApiInfoExtended apiInfo2 = ServiceInfo.ApiInfoExtended.builder().apiId("zowe.apiml.gateway").build();
+            apiml.setApiInfo(Arrays.asList(apiInfo, apiInfo2));
+
+            ApimlInfo info = centralApimlInfoMapper.buildApimlServiceInfo("apiml1", singletonList(serviceOne));
+
+            CentralServiceInfo centralService = info.getServices().get(0);
+            assertThat(centralService.getApiId()).containsOnly("zowe.apiml.apicatalog", "zowe.apiml.gateway");
+        }
+
+        @Test
+        void shouldMapListOfServices() {
+            //bare minimal registration
```

Comment:
is this comment needed?

Reply:
I left it on purpose, to explain that for this case I set only 2 parameters serviceId and status